dashboard/app/lib/user_progress.rb

```diff
@@ -0,0 +1,13 @@
+module UserProgress
+
```

Comment:
A lot of the refactoring I'm working on in https://github.com/code-dot-org/code-dot-org/pull/4716 is going to end up in this module as well.